internal/keppel/manifest.go

```diff
@@ -24,158 +24,235 @@ import (
-	"github.com/docker/distribution/manifest/ocischema"
-	"github.com/docker/distribution/manifest/schema2"
+	"github.com/containers/image/v5/manifest"
+	"github.com/containers/image/v5/types"
+	imageSpecs "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
-//NOTE: We don't enable github.com/docker/distribution/manifest/schema1
-// anymore since it's legacy anyway and the implementation is a lot simpler
-// when we don't have to rewrite manifests between schema1 and schema2.
-
+// TODO: use manifest.Manifest instead!?!?
 // ParsedManifest is an interface that can interrogate manifests about the blobs
 // and submanifests referenced therein.
 type ParsedManifest interface {
 	// FindImageConfigBlob returns the descriptor of the blob containing this
 	// manifest's image configuration, or nil if the manifest does not have an image
 	// configuration.
-	FindImageConfigBlob() *distribution.Descriptor
+	FindImageConfigBlob() *types.BlobInfo
 	// FindImageLayerBlobs returns the descriptors of the blobs containing this
 	// manifest's image layers, or an empty list if the manifest does not have layers.
-	FindImageLayerBlobs() []distribution.Descriptor
+	FindImageLayerBlobs() []manifest.LayerInfo
 	// BlobReferences returns all blobs referenced by this manifest.
-	BlobReferences() []distribution.Descriptor
+	BlobReferences() []manifest.LayerInfo
 	// ManifestReferences returns all manifests referenced by this manifest.
-	ManifestReferences(pf models.PlatformFilter) []manifestlist.ManifestDescriptor
+	ManifestReferences(pf models.PlatformFilter) []imageSpecs.Descriptor
 	// AcceptableAlternates returns the subset of ManifestReferences() that is
 	// acceptable as alternate representations of this manifest. When a client
 	// asks for this manifest, but the Accept header does not match the manifest
 	// itself, the API will look for an acceptable alternate to serve instead.
-	AcceptableAlternates(pf models.PlatformFilter) []manifestlist.ManifestDescriptor
+	AcceptableAlternates(pf models.PlatformFilter) []imageSpecs.Descriptor
+}
+
+var ManifestMediaTypes = []string{
+	manifest.DockerV2ListMediaType,
+	manifest.DockerV2Schema2MediaType,
+	imageSpecs.MediaTypeImageIndex,
+	imageSpecs.MediaTypeImageManifest,
 }
 
 // ParseManifest parses a manifest. It also returns a Descriptor describing the manifest itself.
-func ParseManifest(mediaType string, contents []byte) (ParsedManifest, distribution.Descriptor, error) {
-	m, desc, err := distribution.UnmarshalManifest(mediaType, contents)
-	if err != nil {
-		return nil, distribution.Descriptor{}, err
-	}
-	switch m := m.(type) {
-	case *schema2.DeserializedManifest:
-		return v2ManifestAdapter{m}, desc, nil
-	case *ocischema.DeserializedManifest:
-		return ociManifestAdapter{m}, desc, nil
-	case *manifestlist.DeserializedManifestList:
-		return listManifestAdapter{m}, desc, nil
+func ParseManifest(mediaType string, contents []byte) (ParsedManifest, error) {
+	// update ManifestMediaTypes if any new are added
```

Comment:
This looks like a description of the code, instead of an admonishment, so it will likely be missed at the important moments.
```suggestion
	// WARNING: Please update ManifestMediaTypes if any new are added.
```